### PR TITLE
Bug fix: Polling status using POST-as-GET wherever possible

### PR DESCRIPTION
**Pull request by sam-lord**
Avoid repeating finalize POST request and challenge POST requests by
using POST-as-GET requests instead. Allows for testing with Pebble,
and more correctly follows the spec.

### CODE REVIEW

File: lib/native.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ var promisify = require('util').promisify;
 var resolveTxt = promisify(require('dns').resolveTxt);
 var crypto = require('crypto');
 
+const https = require('https');
+var http01Agent = new https.Agent({
+	rejectUnauthorized: false
```
Comment (coolaj86) [inline lib/native.js:10]: This should be an option or environment variable. The default behavior should not be to allow unsigned, expired, and/or unverifiable certificates.

File: lib/native.js [hunk 2]
```diff
@@ -27,7 +32,8 @@ native._dns01 = function (me, ch) {
 
 native._http01 = function (me, ch) {
 	return new me.request({
-		url: ch.challengeUrl
+		url: ch.challengeUrl,
+		agent: http01Agent
```
Comment (coolaj86) [inline lib/native.js:36]: Monkey patching `native._http01` in cases where an insecure agent is required would also be acceptable.

**Review by coolaj86 — CHANGES_REQUESTED** [review]
Definitely don't want to make requests insecure by default.

The other changes look reasonable. Assuming that you've tested them against Let's Encrypt as well as pebble, these seem fine to me.

**Comment by sam-lord** [comment]
My bad, those changes were not meant for this PR - I was planning on tidying those up once I'd tested that it works for my use case. I'm going to clear those changes from this PR and just leave the original. I'll bring up the http-01 over HTTPS separately

**Comment by sam-lord** [comment]
Reverted the changes mentioned, they were not meant for this PR and were part of some internal testing I was doing :)

We have tested the original changes for this PR for over a week against real LetsEncrypt *and* Pebble.

**Review by coolaj86 — APPROVED** [review]
LGTM

**Comment by coolaj86** [comment]
Rebase / Merge at will.

**Pull request merged**